akka-parsing/src/main/scala-3/akka/parboiled2/support/TailSwitch.scala

```diff
@@ -74,6 +74,15 @@ object TailSwitch {
   type Aux[L <: HList, LI <: HList, T <: HList, TI <: HList, R <: HList, RI <: HList, Out0] =
     TailSwitch[L, T, R] { type Out = Out0 }
 
-  implicit def tailSwitch[L <: HList, T <: HList, R <: HList]
-      : TailSwitch[L, T, R] { type Out = TailSwitch0[L, L, T, T, R, HNil] } = `n/a`
+  implicit def tailSwitch[L <: _ :: _, T <: _ :: _, R <: HList]
+  : TailSwitch[L, T, R] { type Out = TailSwitch0[L, L, T, T, R, HNil] } = `n/a`
+  /// Optimisations to reduce compilation times to something tolerable
```

Comment:
I couldn't figure out solve the general issue, but having some special-case implicits to short-circuit logic in certain common patterns seems to be enough for the usages in akka-http to compile

Reply:
I've been trying to solve this in a more general way but it looks like this would require some fixes/optimizations in the compiler itself.

Reply:
Yeah, I had the same feeling... I poked around a little but it looks like the main issue is that the compiler is 'surprisingly slow' at evaluating the compile-time type reduction. Guessing the culprit could be the type destructuring, but didn't try to add a test-case to the compiler itself...

Reply:
Anyway apologies for stepping on your toes with this one. I got rather carried away testing that the workaround held good for all the cases we encounter in the parsers 😅